lib/state/ui/search-field-middleware.ts

```diff
@@ -10,10 +10,19 @@ export const middleware: S.Middleware = () => {
     const result = next(action);
 
     switch (action.type) {
-      case 'SEARCH':
+      case 'SEARCH': {
         searchFields.forEach((focus) => focus());
+        const model = window.editor.getModel();
+        const range = model.findMatches(action.searchQuery)[0]?.range;
+        if (range) {
+          window.editor.setSelection(range);
+          window.editor.getAction('actions.find').run();
+        } else {
+          window.editor.setSelection();
+          window.editor.getAction('actions.find').run();
+        }
         break;
-
+      }
```

Comment:
this sits a big uneasily with me as another change did before where we split Monaco-specific functionality between the editor and the Electron middleware.

I've considered pulling the Redux store into the editor in order to listen to these events even as a tradeoff, as bad as that would normally seem, to keep all the specifics about Monaco in one place.

`SEARCH` fires on every keystroke in the search box? if so, do we even need to listen here for changes and imperatively update the editor or is it already convenient enough that we can watch for the pure state prop `searchQuery` in  `note-content-editor::getDerivedStateFromProps` and update the editor when it changes?